External json storage

AloeSapling opened the pull request:
Added the ability to export to externally hosted storage for JSON data and import data from that URL with a button, allowing for easier template sharing and potential for self-updating templates


Changes:

- Return boolean in importJson and parseBlueMarble that is true when import / parse succeeds and false if not;
- Updating templatesJson in parseBlueMarble;
- Storing templatesJson in parseBlueMarble;
- Input for url from which to import JSON data;
- Button for importing JSON data + error handling;
- Button for exporting JSON data (via file download) + error handling;
- Style / CSS changes for the new elements;

(Large diffs were caused by the code formatter tool "prettier" formatting whole files)

italysuyai583-code commented:
![b5e3dbb394ae16fd88edc903a726c93f (1)](https://github.com/user-attachments/assets/dfbe5b99-62dd-4783-b5dd-f0bf48cbd1ae)

LittleEndu commented:
Several issues with this PR.

* you should not just decide to run code formatting tools. These decisions should be made by the maintainers of the repository, not contributors
* even so, if you want to suggest using prettier, do so in a separate PR or at the very least in a separate commit. Trying to figure out what changes are functional changes is impossible with the commits here
* you have added the userscript meta file into .gitignore for no reason
* even if everything else is addressed, I find an import/export button combo quite cumbersome to use compared to what I have suggested in #48 
  * with import button the player would need to 1) install tampermonkey 2) install bluemarble 3) download json 4) open wplace 5) find and upload json
  * with the json being hosted the player can skip downloading the json and instead go 3) open wplace+json link 4) import json

the difference doesn't seem like a lot but past experience has shown that keeping things as simple as possible leads to alliances getting more of their members to correctly follow the overlay.

AloeSapling commented:
- You are adding correct about the use of prettier and I apologise for that.
- I did make an issue about adding prettier / code formatting tools alongside other changes, which i plan to resolve myself
- As for the changes to the .gitignore file, I forgot I did that, but I did go over why with swingthevine already
- As far as I see in both cases the user needs to have blueMarble installed, the only real difference being that the link to import from is a url search param instead of being an input
Overall, I agree that adding the ability for the URL of the externally hosted JSON to be a search param would make it even easier to share templates and is in fact a feature I thought to add after I already finished the pull request. With the roadmap for future changes I already know I or somebody else will have to change some things around with how exporting and importing works and I figured that such a feature could be added alongside that.

Hope this addresses all of your issues :)

SwingTheVine commented:
But regardless, the .gitignore change needs to be removed, the changes to the meta file need to be removed, and the prettify changes need to be removed.

(I have specifically and intentionally avoided using a prettifier, so I'm not submitting the PR until it is gone.)

AloeSapling commented:
sounds good, I'll provide a valid PR in a moment
